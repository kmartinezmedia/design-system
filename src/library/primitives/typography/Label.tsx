import React, { FunctionComponent } from 'react';
import { StyleSheet } from 'react-native';
import { Text, TextProps } from './Text';
import {
  fontFamily,
  fontWeight,
  fontSize,
  lineHeight,
} from '@designSystem/theme';

export const Label: FunctionComponent<TextProps> = ({
  children,
  spacing = { bottom: 1 },
  ...textProps
}) => (
  <Text style={styles.Label} spacing={spacing} {...textProps}>
    {children}
  </Text>
);

const styles = StyleSheet.create({
  Label: {
    fontFamily: fontFamily.medium,
    fontWeight: fontWeight.medium,
    fontSize: fontSize.label,
    lineHeight: lineHeight.label,
  },
});
