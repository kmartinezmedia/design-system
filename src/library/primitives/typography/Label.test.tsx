import React from 'react';
import 'react-native';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { Label } from './Label';

describe('Label', () => {
  it('default', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<Label>Hey, lil 🐛</Label>);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
