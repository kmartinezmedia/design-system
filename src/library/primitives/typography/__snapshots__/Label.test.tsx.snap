// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label default 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Graphik-Medium",
        "fontSize": 13,
        "fontWeight": "500",
        "lineHeight": 16,
      },
      Object {
        "paddingBottom": 4,
      },
      Object {
        "color": "#071422",
        "lineHeight": 16,
        "overflow": "visible",
        "textAlign": undefined,
      },
      undefined,
    ]
  }
>
  Hey, lil 🐛
</Text>
`;
